File: Graphs/categories.m

% Categories: Gasoline, Food, Rent, Utilities
% Data for Jan, Feb, Mar, Apr
data = [
    20, 180, 500, 200;  % January
    25, 175, 510, 210;  % February
    30, 350, 520, 230;  % March
    22, 160, 505, 215   % April
];

% Transpose for stacked bar format (each row = category)
data = data';

% Month labels
months = {'Jan', 'Feb', 'Mar', 'Apr'};

% Category labels for legend
category = {'Parking', 'Food', 'Accommodation', 'Gasoline'};

% Create stacked bar chart
figure;
bar(data', 'stacked');

% Add labels
title('Expenses by Month');
xlabel('Month');
ylabel('Amount (£)');
set(gca, 'XTickLabel', months);
legend(category, 'Location', 'northeastoutside');

% Optional: improve appearance
grid on;

File: Graphs/python/categories.py
import matplotlib.pyplot as plt
import numpy as np

# Step 1: Define initial data
months = ['Jan', 'Feb', 'Mar', 'Apr']
gasoline = [20, 25, 30, 22]
food = [180, 175, 350, 160]
rent = [500, 510, 520, 505]
utilities = [200, 210, 230, 215]

# Step 2: Change a variable — e.g., update March food expense
food[2] = 300  # originally 350

# Step 3: Convert data for stacked bar
ind = np.arange(len(months))
width = 0.6

# Plotting stacked bar
p1 = plt.bar(ind, gasoline, width, color='blue', label='Gasoline')
p2 = plt.bar(ind, food, width, bottom=gasoline, color='red', label='Food')
p3 = plt.bar(ind, rent, width, bottom=np.array(gasoline) + np.array(food), color='skyblue', label='Rent')
p4 = plt.bar(ind, utilities, width, bottom=np.array(gasoline) + np.array(food) + np.array(rent), color='green', label='Utilities')

# Labels
plt.ylabel('Amount ($)')
plt.title('Expenses by Month')
plt.xticks(ind, months)
plt.legend(loc='upper right')
plt.grid(axis='y')

# Show the plot
plt.tight_layout()
plt.show()
